File: programmers42861.py
n=4
costs=[[0,1,1],[0,2,2],[1,2,5],[1,3,1],[2,3,8]]

def solution(n, costs):
    answer=0
    costs.sort(key=lambda x: x[2])
    union=[i for i in range(n)]
    
    for start, end, cost in costs:
        if union[start]!=union[end]:
            des=union[end]
            for i in range(n):
                if union[i]==des:
                    union[i]=union[start]
            answer+=cost

    return answer

print(solution(n, costs))

File: programmers42861.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool cmp(vector<int> a, vector<int> b) {
	return a[2] < b[2];
}

int getParent(vector<int>& parent, int x) {
	if (parent[x] == x)
		return x;
	return parent[x] = getParent(parent, parent[x]);
}

void unionParent(vector<int>& parent, int a, int b) {
	a = getParent(parent, a);
	b = getParent(parent, b);
	a < b ? parent[b] = a : parent[a] = b;
}

bool find(vector<int>& parent, int a, int b) {
	a = getParent(parent, a);
	b = getParent(parent, b);
	return a == b;
}

int solution(int n, vector<vector<int>> costs) {
	int answer = 0;
	int max = 0;
	sort(costs.begin(), costs.end(), cmp);
	for (auto a : costs)
		if (max < a[1])
			max = a[1];
	vector<int> parent;
	for (int i = 0; i <= max; i++)
		parent.push_back(i);
	for (auto a : costs) {
		if (!find(parent, a[0], a[1])) {
			answer += a[2];
			unionParent(parent, a[0], a[1]);
		}
	}
	return answer;
}